sunpy/physics/differential_rotation.py

```diff
@@ -84,138 +88,318 @@ def diff_rot(duration: u.s, latitude: u.deg, rot_type='howard', frame_time='side
     drot = diff_rot(interval, heliographic_coordinate.lat.to(u.degree), **diff_rot_kwargs)
 
-    # Rotate the input co-ordinate and update the observer
-    heliographic_rotated = SkyCoord(
-        heliographic_coordinate.lon + drot,
-        heliographic_coordinate.lat,
-        obstime=new_observer_time,
-        observer=new_observer_location,
-        frame=frames.HeliographicStonyhurst)
+    # Rotate the input co-ordinate as seen by the original observer
+    heliographic_rotated = SkyCoord(heliographic_coordinate.lon + drot, heliographic_coordinate.lat,
+                                    obstime=coordinate.obstime, observer=coordinate.observer,
+                                    frame=HeliographicStonyhurst)
 
-    # Return the rotated coordinates to the input coordinate frame
-    return heliographic_rotated.transform_to(coordinate.frame.name)
+    # Calculate where the rotated co-ordinate appears as seen by new observer,
+    # and then transform it into the co-ordinate system of the input
+    # co-ordinate.
+    return heliographic_rotated.transform_to(new_observer).transform_to(coordinate.frame.name)
 
 
-@u.quantity_input
-def _warp_sun_coordinates(xy, smap, dt: u.s, **diffrot_kwargs):
+def _rotate_submap_edge(smap, pixels, observer, **diff_rot_kwargs):
+    """
+    Helper function that is used to calculate where the edge of a rectangular
+    map move to on rotation.  If all the pixels passed in are not on disk and
+    therefore subject to solar differential rotation, the coordinates
+    corresponding to the input pixels are returned.
+
+    Parameters
+    ----------
+    smap : `~sunpy.map.Map`
+        The input map from which the pixel coordinates are calculated.
+
+    pixels : `~astropy.units.Quantity`
+        A Quantity array of shape (M, 2) in pixel units.  Values (:, 0) are the y values of the
+        pixel indices, and values (:, 1) are the x values of the pixel indices.
+
+    observer : `~astropy.coordinates.SkyCoord`
+        The location of the observer.
+
+    diff_rot_kwargs : None, `~dict`
+        Keyword arguments accepted by `~sunpy.physics.differential_rotation.diff_rot`.
+
+    Returns
+    -------
+    coordinates : `~astropy.coordinates.SkyCoord`
+        The coordinates of a rotated edge.
+    """
+    # Coordinates
+    c = smap.pixel_to_world(pixels[:, 1], pixels[:, 0])
+
+    # Only apply solar rotation if all coordinates are on the disk.
+    if np.all(~coordinate_is_on_disk(c, smap.rsun_obs)):
+        coordinates = deepcopy(c)
+    else:
+        coordinates = solar_rotate_coordinate(c, observer=observer, **diff_rot_kwargs)
+    return coordinates
+
+
+def _get_extreme_position(coords, axis, operator=np.nanmax):
+    """
+    Helper function that calculates an extreme position from a list of
+    coordinates.
+
+    Parameters
+    ----------
+    coords : `~list`
+        Each member of the list is a `~astropy.coordinates.SkyCoord`.
+
+    axis :  'Tx', 'Ty'
```

Comment:
What are these? String only inputs that are either Tx or Ty?

```suggestion
    axis :  {'Tx', 'Ty'}
```